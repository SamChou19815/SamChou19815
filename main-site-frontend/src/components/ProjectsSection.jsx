// @flow strict

import type { Node } from 'react';
import React from 'react';
// $FlowFixMe
import Card from '@material-ui/core/Card';
// $FlowFixMe
import CardHeader from '@material-ui/core/CardHeader';
// $FlowFixMe
import CardActions from '@material-ui/core/CardActions';
// $FlowFixMe
import CardContent from '@material-ui/core/CardContent';
// $FlowFixMe
import CardMedia from '@material-ui/core/CardMedia';
// $FlowFixMe
import Icon from '@material-ui/core/Icon';
import ConsoleSection from './ConsoleSection';
import styles from './ProjectsSection.module.css';
import ButtonLink from './ButtonLink';
import TEN from '../assets/projects/ten.png';
import SAMLANG from '../assets/projects/samlang.png';
import Samwise from '../assets/projects/samwise.png';

export default (): Node => (
  <ConsoleSection id="projects" title="active-projects --pretty">
    <div className={styles.ProjectContainer}>
      <Card className={styles.ProjectCard}>
        <CardMedia image={SAMLANG} title="SAMLANG" component="img" />
        <CardHeader avatar={<Icon>code</Icon>} title="SAMLANG" subheader="Programming Language" />
        <CardContent>
          A statically-typed functional programming language with full type inference.
          A research programming language developed by Sam.
        </CardContent>
        <CardActions>
          <ButtonLink href="https://samlang.developersam.com" openInNewTab>
            Official Site
          </ButtonLink>
          <ButtonLink href="https://samlang-demo.developersam.com" openInNewTab>
            Demo
          </ButtonLink>
          <ButtonLink href="https://github.com/SamChou19815/samlang" openInNewTab>
            GitHub Repo
          </ButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <CardMedia image={Samwise} title="Samwise" component="img" />
        <CardHeader avatar={<Icon>code</Icon>} title="Samwise" subheader="Web App" />
        <CardContent>
          {'A Student Planner for Everyone. Designed, developed and maintained by '}
          <a href="https://cornelldti.org" target="_blank" rel="noopener noreferrer">Cornell DTI</a>
          .
        </CardContent>
        <CardActions>
          <ButtonLink href="https://github.com/cornell-dti/samwise" openInNewTab>
            GitHub Repo
          </ButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <CardMedia image={TEN} title="TEN" component="img" />
        <CardHeader avatar={<Icon>code</Icon>} title="TEN" subheader="Game AI" />
        <CardContent>
          Interesting board game with simple rules.
          Powered by an MCTS AI.
        </CardContent>
        <CardActions>
          <ButtonLink href="https://ten.developersam.com" openInNewTab>Demo</ButtonLink>
          <ButtonLink href="https://github.com/SamChou19815/ten" openInNewTab>
            GitHub Repo
          </ButtonLink>
        </CardActions>
      </Card>
    </div>
  </ConsoleSection>
);
