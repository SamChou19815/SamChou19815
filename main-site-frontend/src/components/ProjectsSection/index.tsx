import React, { ReactElement } from 'react';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Icon from '@material-ui/core/Icon';
import ConsoleSection from '../Common/ConsoleSection';
import styles from './index.module.css';
import ButtonLink from '../Common/ButtonLink';
import TEN from '../../assets/projects/ten.png';
import SAMLANG from '../../assets/projects/samlang.png';
import Samwise from '../../assets/projects/samwise.png';

export default (): ReactElement => (
  <ConsoleSection id="projects" title="./active-projects --pretty-print">
    <div className={styles.ProjectContainer}>
      <Card className={styles.ProjectCard}>
        <CardMedia image={SAMLANG} title="SAMLANG" component="img" />
        <CardHeader avatar={<Icon>code</Icon>} title="SAMLANG" subheader="Programming Language" />
        <CardContent>
          A statically-typed functional programming language with full type inference.
          A research programming language developed by Sam.
        </CardContent>
        <CardActions>
          <ButtonLink href="https://samlang.developersam.com" openInNewTab>
            Official Site
          </ButtonLink>
          <ButtonLink href="https://samlang-demo.developersam.com" openInNewTab>
            Demo
          </ButtonLink>
          <ButtonLink href="https://github.com/SamChou19815/samlang" openInNewTab>
            GitHub Repo
          </ButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <CardMedia image={Samwise} title="Samwise" component="img" />
        <CardHeader avatar={<Icon>code</Icon>} title="Samwise" subheader="Web App" />
        <CardContent>
          {'A Student Planner for Everyone. Designed, developed and maintained by '}
          <a href="https://cornelldti.org" target="_blank" rel="noopener noreferrer">Cornell DTI</a>
          .
        </CardContent>
        <CardActions>
          <ButtonLink href="https://samwise.today" openInNewTab>
            App
          </ButtonLink>
          <ButtonLink href="https://github.com/cornell-dti/samwise" openInNewTab>
            GitHub Repo
          </ButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <CardMedia image={TEN} title="TEN" component="img" />
        <CardHeader avatar={<Icon>code</Icon>} title="TEN" subheader="Game AI" />
        <CardContent>
          Interesting board game with simple rules.
          Powered by an MCTS AI.
        </CardContent>
        <CardActions>
          <ButtonLink href="https://ten.developersam.com" openInNewTab>Demo</ButtonLink>
          <ButtonLink href="https://github.com/SamChou19815/ten-golang" openInNewTab>
            GitHub Repo
          </ButtonLink>
        </CardActions>
      </Card>
    </div>
  </ConsoleSection>
);
