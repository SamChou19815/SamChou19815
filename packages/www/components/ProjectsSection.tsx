import React, { ReactElement } from 'react';

import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Divider from '@material-ui/core/Divider';

import projects from '../data/projects';
import ConsoleSection from './Common/ConsoleSection';
import LazyMaterialMedia from './Common/LazyMaterialMedia';
import styles from './ProjectsSection.module.css';

import MaterialButtonLink from 'lib-react/MaterialButtonLink';

const ProjectsSection = (): ReactElement => (
  <ConsoleSection id="projects" title="dev-sam projects">
    <div className={styles.ProjectContainer}>
      {projects.map(({ name, type, media, description, links }) => (
        <Card key={name} className={styles.ProjectCard}>
          <LazyMaterialMedia image={media} title={name} />
          <Divider />
          <CardHeader title={name} subheader={type} />
          <CardContent>{description}</CardContent>
          <CardActions>
            {links.map(({ text, href }) => (
              <MaterialButtonLink key={text} href={href}>
                {text}
              </MaterialButtonLink>
            ))}
          </CardActions>
        </Card>
      ))}
    </div>
  </ConsoleSection>
);

export default ProjectsSection;
