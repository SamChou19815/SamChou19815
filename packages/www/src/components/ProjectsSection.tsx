import React, { ReactElement } from 'react';

import clsx from 'clsx';

import DATASET_PROJECTS from '../data/projects';
import ButtonLink from './Common/ButtonLink';
import CardHeader from './Common/CardHeader';
import ConsoleSection from './Common/ConsoleSection';
import LazyCardMedia from './Common/LazyCardMedia';
import styles from './ProjectsSection.module.css';

const ProjectsSection = (): ReactElement => (
  <ConsoleSection id="projects" title="dev-sam projects">
    <div className={styles.ProjectContainer}>
      {DATASET_PROJECTS.map(({ name, type, media, description, links }) => (
        <div key={name} className={clsx('card', styles.ProjectCard)}>
          <LazyCardMedia image={media} title={name} />
          <CardHeader title={name} subheader={type} />
          <div className="card__body">{description}</div>
          <div className="card__footer">
            <div className="button-group button-group--block">
              {links.map(({ text, href }) => (
                <ButtonLink key={text} href={href}>
                  {text}
                </ButtonLink>
              ))}
            </div>
          </div>
        </div>
      ))}
    </div>
  </ConsoleSection>
);

export default ProjectsSection;
