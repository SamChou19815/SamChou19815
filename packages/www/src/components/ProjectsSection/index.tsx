import React, { ReactElement } from 'react';

import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Divider from '@material-ui/core/Divider';
import Code from '@material-ui/icons/Code';
import MaterialButtonLink from 'lib-react/MaterialButtonLink';

import Docusaurus from '../../assets/projects/docusaurus.png';
import MiniReact from '../../assets/projects/mini-react.png';
import SAMLANG from '../../assets/projects/samlang.png';
import Samwise from '../../assets/projects/samwise.png';
import Tasks from '../../assets/projects/tasks.png';
import TEN from '../../assets/projects/ten.png';
import ConsoleSection from '../Common/ConsoleSection';
import LazyMaterialMedia from '../Common/LazyMaterialMedia';
import styles from './index.module.css';

export default (): ReactElement => (
  <ConsoleSection id="projects" title="projects --active">
    <div className={styles.ProjectContainer}>
      <Card className={styles.ProjectCard}>
        <LazyMaterialMedia image={SAMLANG} title="SAMLANG" />
        <Divider />
        <CardHeader avatar={<Code />} title="SAMLANG" subheader="Programming Language" />
        <CardContent>
          A statically-typed functional programming language with full type inference. A research
          programming language developed by Sam.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="https://samlang.developersam.com">
            Official Site
          </MaterialButtonLink>
          <MaterialButtonLink href="https://samlang-demo.developersam.com">Demo</MaterialButtonLink>
          <MaterialButtonLink href="https://github.com/SamChou19815/samlang">
            GitHub Repo
          </MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <LazyMaterialMedia image={Docusaurus} title="Docusaurus" />
        <Divider />
        <CardHeader avatar={<Code />} title="Docusaurus" subheader="Open source contributions" />
        <CardContent>
          A static documentation site generator developed by Facebook open source. I am one of the
          top 20 contributors.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="https://v2.docusaurus.io">Official Site</MaterialButtonLink>
          <MaterialButtonLink href="https://github.com/facebook/docusaurus">
            GitHub Repo
          </MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <LazyMaterialMedia image={MiniReact} title="mini-react" />
        <Divider />
        <CardHeader avatar={<Code />} title="mini-react" subheader="Framework" />
        <CardContent>
          A simplified version of React runtime created from scratch. It has support for the{' '}
          <code>useState</code> hook and <code>useEffect</code> hook.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="https://mini-react.developersam.com">
            Demo Site
          </MaterialButtonLink>
          <MaterialButtonLink href="https://github.com/SamChou19815/mini-react">
            GitHub Repo
          </MaterialButtonLink>
          <MaterialButtonLink href="/build-simplified-react.pdf">Slides</MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <LazyMaterialMedia image={Samwise} title="Samwise" />
        <Divider />
        <CardHeader avatar={<Code />} title="Samwise" subheader="Web App" />
        <CardContent>
          A Student Planner for Everyone. Designed, developed and maintained by&nbsp;
          <a href="https://cornelldti.org">Cornell DTI</a>.
        </CardContent>
        <CardActions>
          <MaterialButtonLink href="https://samwise.today">App</MaterialButtonLink>
          <MaterialButtonLink href="https://github.com/cornell-dti/samwise">
            GitHub Repo
          </MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <LazyMaterialMedia image={Tasks} title="Tasks" />
        <Divider />
        <CardHeader avatar={<Code />} title="Tasks" subheader="Web App" />
        <CardContent>A planner with drag and drop and dependency analysis.</CardContent>
        <CardActions>
          <MaterialButtonLink href="https://tasks.developersam.com">App</MaterialButtonLink>
        </CardActions>
      </Card>
      <Card className={styles.ProjectCard}>
        <LazyMaterialMedia image={TEN} title="TEN" />
        <Divider />
        <CardHeader avatar={<Code />} title="TEN" subheader="Game AI" />
        <CardContent>Interesting board game with simple rules. Powered by an MCTS AI.</CardContent>
        <CardActions>
          <MaterialButtonLink href="https://ten.developersam.com">Demo</MaterialButtonLink>
          <MaterialButtonLink href="https://github.com/SamChou19815/ten-golang">
            GitHub Repo
          </MaterialButtonLink>
        </CardActions>
      </Card>
    </div>
  </ConsoleSection>
);
