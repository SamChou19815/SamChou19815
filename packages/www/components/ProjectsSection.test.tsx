import React from 'react';

import renderer from 'react-test-renderer';

import ProjectsSection from './ProjectsSection';

it('ProjectsSection matches snapshot.', () => {
  const tree = renderer.create(<ProjectsSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
