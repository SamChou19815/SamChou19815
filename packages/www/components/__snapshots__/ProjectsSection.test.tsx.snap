// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsSection matches snapshot. 1`] = `
<section
  id="projects"
>
  <h3
    className="Title"
  >
    <code>
      $Â 
      dev-sam projects
    </code>
  </h3>
  <div
    className="ProjectContainer"
  >
    <div
      className="card ProjectCard"
    >
      <div
        className="card__image"
      >
        <img
          alt="SAMLANG"
          loading="lazy"
          src="/projects/samlang.png"
          title="SAMLANG"
        />
      </div>
      <div
        className="card__header"
      >
        <div
          className="avatar"
        >
          <div
            className="avatar__intro"
          >
            <h4
              className="avatar__name"
            >
              SAMLANG
            </h4>
            <small
              className="avatar__subtitle"
            >
              Programming Language
            </small>
          </div>
        </div>
      </div>
      <div
        className="card__body"
      >
        A statically-typed functional programming language with full type inference. A research programming language developed by Sam.
      </div>
      <div
        className="card__footer"
      >
        <div
          className="button-group button-group--block"
        >
          <a
            className="button button--link"
            href="https://samlang.developersam.com"
          >
            DOCS
          </a>
          <a
            className="button button--link"
            href="https://samlang.developersam.com/demo"
          >
            DEMO
          </a>
          <a
            className="button button--link"
            href="https://github.com/SamChou19815/samlang"
          >
            GITHUB
          </a>
        </div>
      </div>
    </div>
    <div
      className="card ProjectCard"
    >
      <div
        className="card__image"
      >
        <img
          alt="Docusaurus"
          loading="lazy"
          src="/projects/docusaurus.png"
          title="Docusaurus"
        />
      </div>
      <div
        className="card__header"
      >
        <div
          className="avatar"
        >
          <div
            className="avatar__intro"
          >
            <h4
              className="avatar__name"
            >
              Docusaurus
            </h4>
            <small
              className="avatar__subtitle"
            >
              Open source contributions
            </small>
          </div>
        </div>
      </div>
      <div
        className="card__body"
      >
        A static documentation site generator developed by Facebook open source. I am one of the top 20 contributors.
      </div>
      <div
        className="card__footer"
      >
        <div
          className="button-group button-group--block"
        >
          <a
            className="button button--link"
            href="https://v2.docusaurus.io"
          >
            DOCS
          </a>
          <a
            className="button button--link"
            href="https://github.com/facebook/docusaurus"
          >
            GITHUB
          </a>
        </div>
      </div>
    </div>
    <div
      className="card ProjectCard"
    >
      <div
        className="card__image"
      >
        <img
          alt="mini-react"
          loading="lazy"
          src="/projects/mini-react.png"
          title="mini-react"
        />
      </div>
      <div
        className="card__header"
      >
        <div
          className="avatar"
        >
          <div
            className="avatar__intro"
          >
            <h4
              className="avatar__name"
            >
              mini-react
            </h4>
            <small
              className="avatar__subtitle"
            >
              Framework
            </small>
          </div>
        </div>
      </div>
      <div
        className="card__body"
      >
        A simplified version of React runtime with useState and useEffect hook, built from Scratch.
      </div>
      <div
        className="card__footer"
      >
        <div
          className="button-group button-group--block"
        >
          <a
            className="button button--link"
            href="https://mini-react.developersam.com"
          >
            DEMO
          </a>
          <a
            className="button button--link"
            href="https://github.com/SamChou19815/mini-react"
          >
            GITHUB
          </a>
          <a
            className="button button--link"
            href="/build-simplified-react.pdf"
          >
            SLIDES
          </a>
        </div>
      </div>
    </div>
    <div
      className="card ProjectCard"
    >
      <div
        className="card__image"
      >
        <img
          alt="Samwise"
          loading="lazy"
          src="/projects/samwise.png"
          title="Samwise"
        />
      </div>
      <div
        className="card__header"
      >
        <div
          className="avatar"
        >
          <div
            className="avatar__intro"
          >
            <h4
              className="avatar__name"
            >
              Samwise
            </h4>
            <small
              className="avatar__subtitle"
            >
              Web App
            </small>
          </div>
        </div>
      </div>
      <div
        className="card__body"
      >
        A Student Planner for Everyone. Designed, developed and maintained by Cornell DTI.
      </div>
      <div
        className="card__footer"
      >
        <div
          className="button-group button-group--block"
        >
          <a
            className="button button--link"
            href="https://samwise.today"
          >
            APP
          </a>
          <a
            className="button button--link"
            href="https://github.com/cornell-dti/samwise"
          >
            GITHUB
          </a>
        </div>
      </div>
    </div>
    <div
      className="card ProjectCard"
    >
      <div
        className="card__image"
      >
        <img
          alt="Wiki"
          loading="lazy"
          src="/projects/wiki.png"
          title="Wiki"
        />
      </div>
      <div
        className="card__header"
      >
        <div
          className="avatar"
        >
          <div
            className="avatar__intro"
          >
            <h4
              className="avatar__name"
            >
              Wiki
            </h4>
            <small
              className="avatar__subtitle"
            >
              Web App
            </small>
          </div>
        </div>
      </div>
      <div
        className="card__body"
      >
        Documentation for this monoropo and notes from Developer Sam.
      </div>
      <div
        className="card__footer"
      >
        <div
          className="button-group button-group--block"
        >
          <a
            className="button button--link"
            href="https://wiki.developersam.com"
          >
            WEBSITE
          </a>
          <a
            className="button button--link"
            href="https://github.com/SamChou19815/website/tree/master/packages/wiki"
          >
            GITHUB
          </a>
        </div>
      </div>
    </div>
    <div
      className="card ProjectCard"
    >
      <div
        className="card__image"
      >
        <img
          alt="TEN"
          loading="lazy"
          src="/projects/ten.png"
          title="TEN"
        />
      </div>
      <div
        className="card__header"
      >
        <div
          className="avatar"
        >
          <div
            className="avatar__intro"
          >
            <h4
              className="avatar__name"
            >
              TEN
            </h4>
            <small
              className="avatar__subtitle"
            >
              Game AI
            </small>
          </div>
        </div>
      </div>
      <div
        className="card__body"
      >
        Interesting board game with simple rules. Powered by an MCTS AI.
      </div>
      <div
        className="card__footer"
      >
        <div
          className="button-group button-group--block"
        >
          <a
            className="button button--link"
            href="https://ten.developersam.com"
          >
            DEMO
          </a>
          <a
            className="button button--link"
            href="https://github.com/SamChou19815/ten-golang"
          >
            GITHUB
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
